engine/crates/federated-graph/src/from_sdl/value.rs

```diff
@@ -0,0 +1,51 @@
+use super::{ast, executable_ast};
+
+pub(super) fn executable_value_to_type_system_value(value: executable_ast::Value<'_>) -> ast::Value<'_> {
+    match value {
+        executable_ast::Value::Variable(v) => ast::Value::Variable(v),
+        executable_ast::Value::Int(v) => ast::Value::Int(v),
+        executable_ast::Value::Float(v) => ast::Value::Float(v),
+        executable_ast::Value::String(v) => ast::Value::String(v),
+        executable_ast::Value::Boolean(v) => ast::Value::Boolean(v),
+        executable_ast::Value::Enum(v) => ast::Value::Enum(v),
+        executable_ast::Value::List(v) => {
+            ast::Value::List(v.into_iter().map(executable_value_to_type_system_value).collect())
+        }
+        executable_ast::Value::Object(v) => ast::Value::Object(
+            v.into_iter()
+                .map(|(name, value)| (name, executable_value_to_type_system_value(value)))
+                .collect(),
+        ),
+        executable_ast::Value::Null => ast::Value::Null,
+    }
```

Comment:
I really need to unify and/or otherwise tidy the Value stuff up in the parser sometime...

Reply:
I wonder about Value::Variable. I can't think of a scenario where it would be in a SDL document. That would be a reason to have two enums.

Reply:
Yeah I think two enums makes sense, but I probably want something closer to the `Value`/`ConstValue` approach that async-graphql uses.  Haven't sat down and given it proper thought though.

One thing I do like about the current separation is that they use different `ValueId` types, since that should help prevent mistakes if you end up needing to use those ids.  Not sure how likely that is really though.

I also kind of wanted to refactor the `Value`s themselves to be a bit more lazy - currently calling `Argument::value` (or whatever) could be kind of expensive since it constructs the whole heirarchy of objects & lists etc...

Lots to do around Values basically - probably the bit I'm the least happy with currently.